File: backend/src/main/java/com/danhooper/homepage/model/RssArticle.java
package com.danhooper.homepage.model;

import com.rometools.rome.feed.synd.SyndContent;

import java.util.List;

public class RssArticle {
    private final String title;
    private final List<SyndContent> content;
    private final String description;
    private final String url;

    public RssArticle(String title, List<SyndContent> content, String description, String url) {
        this.title = title;
        this.content = content;
        this.description = description;
        this.url = url;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getUrl() {
        return url;
    }

    public List<SyndContent> getContent() {
        return content;
    }
}

File: frontend/src/models/RssArticle.ts
export interface IRssArticle {
    title: string;
    description: string;
    url: string;
    content: SyndContent[];
}

export interface SyndContent {
    type: string;
    value: string;
}

export class RssArticle implements IRssArticle {
    title: string;
    description: string;
    url: string;
    content: SyndContent[];
    longDescription: string;

    constructor(article: IRssArticle) {
        this.title = article.title;
        this.description = article.description;
        this.url = article.url;
        this.content = article.content;

        const htmlContent = this.content.find(
            (content) => content.type === 'html',
        );
        this.longDescription = htmlContent
            ? htmlContent.value
            : this.description;
    }
}
